### ISSUE TITLE
SSM wrong output when referencing secretsmanager

### ISSUE THREAD

**Issue by felipiberdun** [localstack/localstack#3496]
# Type of request: This is a bug report

[x] bug report
[ ] feature request

# Detailed description

When I create a new secret in secrets manager and try to retrieve it using ssm get-parameters request, I'm receiving a parse error from the aws java sdk. The request to secrets manager was introduced by solving [this issue](https://github.com/localstack/localstack/issues/3128) and it's working expect that it isn't outputing the proper data type.

## Expected behavior

According to [AWS documentation](https://docs.aws.amazon.com/systems-manager/latest/APIReference/API_Parameter.html#systemsmanager-Type-Parameter-SourceResult) and its service behaviour, the property "SourceResult" should be a string containing the reference of secrets manager.

Response example:
```
{
    "Parameters": [
        {
            "Name": "/aws/reference/secretsmanager/dbSecret",
            "Type": "SecureString",
            "Value": "{\"password\": \"MY_SECRET_PWD\"}",
            "Version": 0,
            "SourceResult": "{\"ARN\":\"arn:aws:secretsmanager:us-east-1:ACCOUNT_ID:secret:dbSecret-WaRQa\",\"name\":\"dbSecret\",\"versionId\":\"7a872371-ff6c-4e6f-b966-747d235b660f\",\"secretString\":\"{\"password\": \"MY_SECRET_PWD\"}\",\"versionStages\":[\"AWSCURRENT\"],\"createdDate\":\"Nov 10, 2020, 10:05:44 PM\"}",
            "LastModifiedDate": "2020-11-10T16:05:44.026000-06:00",
            "ARN": "arn:aws:secretsmanager:us-east-1:ACCOUNT_ID:secret:dbSecret-WaRQa"
        }
    ],
    "InvalidParameters": []
}
```

## Actual behavior

The java sdk is throwing the following error:

`com.amazonaws.SdkClientException: Unable to unmarshall response (We expected a VALUE token but got: START_OBJECT).`

# Steps to reproduce

## Command used to start LocalStack

Started with docker-compose with the following service settings:

```
version: "3.7"
services:
  localstack:
    image: localstack/localstack:0.12.5
    environment:
      SERVICES: dynamodb,secretsmanager,s3,ssm,sns,sqs,sts
      HOSTNAME_EXTERNAL: localstack
```

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)

`awslocal secretsmanager create-secret --name dbSecret --secret-string "{\"password\": \"MY_SECRET_PWD\"}"`

`awslocal ssm get-parameters --names "/aws/reference/secretsmanager/dbSecret" --with-decryption`

```
{
    "Parameters": [
        {
            "Name": "/aws/reference/secretsmanager/dbSecret",
            "Type": "SecureString",
            "Value": "{\"password\": \"MY_SECRET_PWD\"}",
            "SourceResult": {
                "ARN": "arn:aws:secretsmanager:us-east-1:000000000000:secret:dbSecret-WaRQa",
                "Name": "dbSecret",
                "VersionId": "ddd9da6b-0f7f-43c1-bb8a-c8191f72ef5f",
                "SecretString": "{\"password\": \"MY_SECRET_PWD\"}",
                "VersionStages": [
                    "AWSCURRENT"
                ],
                "CreatedDate": 1611161246.0
            },
            "LastModifiedDate": "2021-01-20T10:47:26-06:00"
        }
    ],
    "InvalidParameters": []
}
```

The call is outputing the SourceResult but as an object instead of if stringfied.

**Comment by felipiberdun** [comment]
I forgot to assign this issue to me 🤦. Sorry about that.

**Comment by whummer** [comment]
This is fixed in #3499 and now available in the `latest` Docker image. Thanks again!
